polkadot/node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -491,23 +492,39 @@ where
 		.await
 		.map_err(Error::CancelledAvailabilityCores)??;
 
-	let para_now = match polkadot_node_subsystem_util::signing_key_and_index(&validators, keystore)
-		.and_then(|(_, index)| polkadot_node_subsystem_util::find_validator_group(&groups, index))
-	{
-		Some(group) => {
-			let core_now = rotation_info.core_for_group(group, cores.len());
+	let core_now = if let Some(group) =
+		polkadot_node_subsystem_util::signing_key_and_index(&validators, keystore).and_then(
+			|(_, index)| polkadot_node_subsystem_util::find_validator_group(&groups, index),
+		) {
+		rotation_info.core_for_group(group, cores.len())
+	} else {
+		gum::trace!(target: LOG_TARGET, ?relay_parent, "Not a validator");
+		return Ok(())
+	};
 
+	let para_now = match fetch_claim_queue(sender, relay_parent).await.map_err(Error::Runtime)? {
+		Some(claim_queue) => {
+			// Runtime supports claim queue - use it
+			//
+			// `relay_parent_mode` is not examined here because if the runtime supports claim queue
+			// then it supports async backing params too (`ASYNC_BACKING_STATE_RUNTIME_REQUIREMENT`
+			// < `CLAIM_QUEUE_RUNTIME_REQUIREMENT`).
+			//
+			// Implementation note: here it might be better to fetch the whole claim queue for the
+			// core and use it as `paras_now`. Practical tests however showed that there is no
+			// benefit in doing this so to avoid unnecessary complexity we just only get the first
+			// claim from the queue.
+			claim_queue.get_claim_for(core_now, 0)
```

Comment:
How can this not make a difference. If you only look at entry `0`, the claim has absolutely no advantage over `next_up_on_available`. This change should essentially be a noop then. I guess it can work in tests if the validation of PoVs is real quick and networking is fast (usually in tests).

Reply:
(but then it should have equally worked with `next_up_on_available`.

Reply:
the difference is that it wasn't looking at `next_up_on_available` previously. It was only looking at the para id of the occupied core (which is obviously an issue and only works for bulk coretime)

Reply:
this now leverages the full claim queue, resolving